### PR TITLE
Add several more useful ATs to help out modders

### LINKED ISSUE #1301: TransparentBlock is protected
**Minecraft Version:** {Minecraft version}
1,21

**NeoForge Version:** {NeoForge version. *Version number, not latest/rb*}
neo_version=21.0.86-beta

**Logs:** {Link(s) to GitHub Gist with full latest.log and/or crash report}
no logs
**Steps to Reproduce:**
Try to register a **TransparentBlock** It is protected in neoForge

**Description of issue:**

forge 1.21
```
 public TransparentBlock(BlockBehaviour.Properties p_312723_) {
        super(p_312723_);
    }
```

neoForge 1.21
```
    protected TransparentBlock(BlockBehaviour.Properties p_309186_) {
        super(p_309186_);
    }

### PR DESCRIPTION
**Pull request by TelepathicGrunt**
Adds several new ATs that I saw suggested by modders. Here's the reasoning:

> **LivingEntity#getCurrentSwingDuration** - Basically the only thing required for people to properly use swinging for attacks of any kind for modded entities. it should be a standard approach for doing attack timings but because it's private it doesn't get used properly

> **LivingEntity#getHitBox** - ??? pretty standard method lol. More relevant than ever now that mc uses it for all melee attacking checks.

>**BiomeGenerationSettings <init>** - the BiomeGenerationSettings constructor should be public. It's impossible to datagen [this biome](https://github.com/TerraFirmaCraft/TerraFirmaCraft/blob/1.20.x/src/main/resources/data/tfc/worldgen/biome/plains.json) using the builder (without AT)

> **ItemModelGenerators#GENERATED_TRIM_MODELS**, **ItemModelGenerators$TrimModelData**, and **ItemModelGenerators$TrimModelData <init>** - Maybe the datagen stuff for trims.

> **Most Particle constructors** - Noticed at both me and other modders were ATing particle constructors to make variants. So just make all of MC's private ones be protected so we all can extend

> **Blocks#always**, **Blocks#never**, **Blocks#ocelotOrParrot** - Seems to be AT'd by people to get something more readable than a short lambda.

> **Item#getPlayerPOVHitResult** - A helper static method that people try to use often (even me iirc). Helps us match how MC does their POV hit result calculations.

> **ShapedRecipe#result** - Set to protected from package-private to allow for easier xmap codec usage and serialization without exposing result entirely to outside consumers. Outside folks still should use the RegistryAccess getter. 

> **PoweredRailBlock's second constructor**, **TransparentBlock#init**, **WaterloggedTransparentBlock#init**, **WeatheringCopperDoorBlock#init**, **WeatheringCopperGrateBlock#init**, **WeatheringCopperTrapDoorBlock #init** - Missing block ATs we should have. Our patched constructor into PoweredRailBlock was also protected for some reason

These ATs seems reasonable to me. Any objections or additional ATs to add?

### CODE REVIEW

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]
Can you add a comment where the particle ATs start and end to make sure that we don't end up adding stuff in the middle? After looking again, there seems to be #group and #endgroup comments. I'd make them more clear like `# Below you will find particle constructor ATs`, and I'd add some empty newlines in between the start comment and the entry before.

Side note for the future, we really should AT some constructors automatically.

**Comment by alcatrazEscapee** [comment]
`Item.getPlayerPOVHitResult` would be nice, it's a protected static method (and so can be hacked around i.e. [like so](https://github.com/TerraFirmaCraft/TerraFirmaCraft/blob/7da7c1c224780d094ae9461420778ba86b6939c0/src/main/java/net/dries007/tfc/util/Helpers.java#L1783-L1792)), but it's just a convienient helper method

**Comment by TelepathicGrunt** [comment]
@alcatrazEscapee Good catch. I recall wanting that before. Added AT for it

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]
This PR has copied the AT from #1270. I would still like to see a use case which cannot be accomplished (in a much more safe manner) using `getResultItem`, with either a registry access from the codec or an empty one (which will crash recipes that actually need it instead of returning empty or wrong stacks)

**Comment by TelepathicGrunt** [comment]
Closes https://github.com/neoforged/NeoForge/issues/1301

File: src/main/resources/META-INF/accesstransformer.cfg [hunk 1]
```diff
@@ -407,9 +428,46 @@ private-f net.minecraft.world.level.storage.loot.LootPool rolls # rolls
...
 public com.mojang.blaze3d.vertex.VertexFormatElement$Usage$SetupState
-public net.minecraft.world.level.block.LiquidBlock fluid
-public net.minecraft.world.item.BucketItem content
-public net.minecraft.world.item.ItemStack addToTooltip(Lnet/minecraft/core/component/DataComponentType;Lnet/minecraft/world/item/Item$TooltipContext;Ljava/util/function/Consumer;Lnet/minecraft/world/item/TooltipFlag;)V
-public net.minecraft.world.item.SwordItem createToolProperties()Lnet/minecraft/world/item/component/Tool;
-public net.minecraft.world.level.block.CropBlock hasSufficientLight(Lnet/minecraft/world/level/LevelReader;Lnet/minecraft/core/BlockPos;)Z # hasSufficientLight
-public net.minecraft.world.item.crafting.ShapedRecipe pattern # pattern
\ No newline at end of file
+
+# Below you will find particle constructor ATs
```
Comment (marchermans) [inline src/main/resources/META-INF/accesstransformer.cfg:432]: I was thinking about this, during the last few days:
I think we might want to introduce a <init>:all mode where all constructors in the class and all of its inheriting types are forced to the visibility set or higher.

That way we will ensure that we won't miss a new particle in the future?

**Review by marchermans — COMMENTED** [review]
See my comment on the Particle stuff.

Also did you remove the ShapedRecipe#result stuff?

**Comment by TelepathicGrunt** [comment]
@marchermans result was removed here: https://github.com/neoforged/NeoForge/pull/1244/commits/591929ec7f8daaaae3ff2bf6d9d017b1535448d1

File: src/main/resources/META-INF/accesstransformer.cfg (hunk 1 above)
Comment (TelepathicGrunt) [reply src/main/resources/META-INF/accesstransformer.cfg:432]: That's beyond my skillset as I wouldn't know where to begin or how to add that feature to ATs. Could it be done after this PR? Like add the feature, then go back and clean up our ATs? 

if it is setup to ignore abstract classes, we can use it to clean up our Block constructor ATs too

Comment (marchermans) [reply src/main/resources/META-INF/accesstransformer.cfg:432]: Yeah sure, just wanted to get your feedback on the idea.

**Review by marchermans — APPROVED** [review]

**Review by Matyrobbrt — APPROVED** [review]

**Pull request merged**